chef.py

```diff
@@ -72,23 +106,194 @@ def construct_channel(self, **kwargs):
+                        title=text, thumbnail=img)
+
+            fetch_assessment_topic_items(driver, topic_node, url, thumbnail=img)
+            parent_node.add_child(topic_node)
+
+
+def _title_exercise(topic_title, first_item, last_item):
+    return "%s %s-%s" % (topic_title, first_item, last_item)
+
+
+QUESTIONS_PER_EXERCISE = 5
+
+def fetch_assessment_topic_items(driver, topic_node, topic_url, thumbnail=None):
+    """Fetch the individual assessment items for a given topic.
+
+    Groups every 5 assessments into an exercise node.
+    """
+    next_item_url = topic_url
+    item_count = 0
+
+    while next_item_url:
+        driver.get(next_item_url)
+        current_url = driver.current_url
+        item_id = current_url.split('/')[-1]
+
+        print('  Fetching question %s (%s)' % (item_count + 1, current_url))
+
+        # Create exercise node, grouping together every 5 questions.
+        if item_count % QUESTIONS_PER_EXERCISE == 0:
+            first_item_index_in_exercise = item_count
+            exercise_title = _title_exercise(topic_node.title, item_count + 1,
+                    item_count + QUESTIONS_PER_EXERCISE)
+            exercise_node = nodes.ExerciseNode(source_id=item_id,
+                    title=exercise_title, license=LICENSE, thumbnail=thumbnail,
+                    exercise_data={'randomize': False})
+            topic_node.add_child(exercise_node)
+
+        # Now try to convert the page HTML into an assessment item, retrying
+        # on error, and then skipping any missing images after a few failed
+        # retries.
+        for i in range(0, 4):
+            try:
+                page_html = get_generated_html_from_driver(driver)
+                question, next_item_url = fetch_assessment_item(page_html, item_id)
+                break
+            except Exception as e:
+                wait_time = (2 ** i)
+                print("Got an error, retrying after a wait of %s seconds. "
+                        "Error was: %s" % (wait_time, str(e)))
+                driver.get(current_url)
+                time.sleep(wait_time)
+                exception = e
+        else:
+            print("Going to try skipping any missing images")
+            page_html = get_generated_html_from_driver(driver)
+            question, next_item_url = fetch_assessment_item(page_html, item_id,
+                    skip_missing_images=True)
+
+        exercise_node.add_question(question)
+        item_count += 1
+
+    # Re-title the exercise, given that this is the last exercise in the topic,
+    # which may not contain up to 5 items. (e.g. re-title it "Genetics 10-12")
+    exercise_node.title = _title_exercise(topic_node.title,
+            first_item_index_in_exercise + 1, item_count)
+
+
+def _process_text_into_markdown(container_node, skip_missing_images):
+    markdown_text = ''
+
+    for node in container_node.children:
+        image = node.select_one('.models-media-Image')
```

Comment:
Is it possible there are multiple images in a question?

If it's always a single image, could add something like `assert len(node.find_all('.models-media-Image'))==1, 'multiple images found'`

Reply:
Nevermind, I saw some questions with multiple images so I guess the "all images" iteration happens when you go over `container_node.children`.

Reply:
Yep, that's the assumption. But good point on the `assert` — in case the website is changed and this assumption no longer holds! Will add it in.